src/dialog-editor/services/dialogValidationService.ts

```diff
@@ -0,0 +1,66 @@
+import {__} from '../../common/translateFunction';
+import * as _ from 'lodash';
+
+export default class DialogValidationService {
+  public invalid: any = {};
+
+  /** @ngInject */
+  constructor(private DialogEditor: any) {
+  }
+
+  /**
+   * Run validations across each dialog elements.
+   * @memberof DialogValidationService
+   * @function dialogIsValid
+   */
+  public dialogIsValid() {
+    let validators = {
```

Comment:
Hm, should this be hardcoded here?

I'm not coming up with any good examples of wanting to add custom validations to the dialog *editor* so maybe the answer is yes :). But is it? :)

Reply:
It would be nice to add new validators, so @romanblanco care to rewrite this like this?
```Typescript
export default class DialogValidationService {
  private validators: any;
  constructor(private DialogEditor: any) {
    this.validators = {
      dialog: DialogValidationService.initDialogValidators(),
      tabs: DialogValidationService.initTabsValidators(),
      groups: DialogValidationService.initGroupsValidators()
      fields: DialogValidationService.initFieldsValidators()
    }
  }
  private static initGroupsValidators() {
    return [
      group => ({ status: ! _.isEmpty(group.label),
                    errorMessage: __('Dialog box needs to have a name') }),
      group => ({ status: group.dialog_fields.length > 0,
                    errorMessage: __('Dialog box needs to have at least one element') })
    ]
  }
  private static initFieldsValidators() {
    return [
      field => ({ status: ! _.isEmpty(field.name),
                    errorMessage: __('Dialog element needs to have a name') }),
      field => ({ status: ! _.isEmpty(field.label),
                    errorMessage: __('Dialog element needs to have a label') })
    ]
  }
  private static initTabsValidators() {
    return [
      tab => ({ status: ! _.isEmpty(tab.label),
                  errorMessage: __('Dialog tab needs to have a name') }),
      tab => ({ status: tab.dialog_groups.length > 0,
                  errorMessage: __('Dialog tab needs to have at least one box') })

    ]
  }
  private static initDialogValidators() {
    return [
      dialog => ({ status: ! _.isEmpty(dialog.label),
                     errorMessage: __('Dialog needs to have a label') }),
      dialog => ({ status: dialog.dialog_tabs.length > 0,
                     errorMessage: __('Dialog needs to have at least one tab') })
    ]
  }
}
```